### ISSUE TITLE
Decoder normalization is being applied to the wrong dimension? 

### ISSUE THREAD

**Issue by jordansauce** [ApolloResearch/e2e_sae#28]
## Bug
[in sparsifiers.py]

The decoder weight has shape `[n_dict_components  input_size]`:
```python
        self.decoder = nn.Linear(n_dict_components, input_size, bias=False)
```
So when it gets orthogonalized, assuming `input_size <= n_dict_components`, the number of orthonormal vectors will be `input_size`. 
Each vector `self.decoder.weight[:,i]` will be orthonormal:  `self.decoder.weight[:,i] @ self.decoder.weight[:,i].T = 1` - so normalized over `dim=1`. 
```python
        # Initialize the decoder weights orthogonally
        nn.init.orthogonal_(self.decoder.weight)
```
```python 
torch.norm(sae.decoder.weight.data, dim=1)
> tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```
The same is not true for `dim=0`:
```python 
torch.norm(sae.decoder.weight.data, dim=0)
> tensor([0.6255, 0.5160, 0.8770, 0.7434, 0.7691, 0.8066, 0.6545, 0.8680, 0.6710,
        0.5591, 0.5021, 0.6295, 0.8675, 0.6582, 0.7962, 0.6448, 0.6287, 0.5879,
        0.8286, 0.7230])
```

However in the forward pass, we normalize over `dim=0`:
```python
    def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
[...]
        # Apply unit norm constraint to the decoder weights
        self.decoder.weight.data = nn.functional.normalize(self.decoder.weight.data, dim=0)
```

## Fix
change `dim=0` to `dim=1` in [sparsify.py line 29](https://github.com/ApolloResearch/sparsify/blob/ce4174b88d73ac255ea3288727ef1b7a5a7dd913/sparsify/models/sparsifiers.py#L29)

Unless this is the intended way around - in which case the initialization should be changed.

[Joseph has it on `dim=1`](https://github.com/jbloomAus/mats_sae_training/blob/f3fe937303f0eb54e83771d1b307c3f09d1145fd/sae_training/sparse_autoencoder.py#L51C1-L59C80):
```python
        self.W_dec = nn.Parameter(
            torch.nn.init.kaiming_uniform_(
                torch.empty(self.d_sae, self.d_in, dtype=self.dtype, device=self.device)
            )
        )

        with torch.no_grad():
            # Anthropic normalize this to have unit columns
            self.W_dec.data /= torch.norm(self.W_dec.data, dim=1, keepdim=True)
```
